src/common/filter/Expressions.h

```diff
@@ -209,6 +210,56 @@ class Expression {
         return std::abs(left - right) < EPSILON;
     }
 
+    // Procedures used to do type casting
+    static std::string toString(const VariantType &value) {
+        char buf[1024];
+        switch (value.which()) {
+            case 0:
+                return folly::to<std::string>(boost::get<int64_t>(value));
+            case 1:
+                return folly::to<std::string>(boost::get<double>(value));
+            case 2:
+                snprintf(buf, sizeof(buf), "%s", boost::get<bool>(value) ? "true" : "false");
+                return buf;
+            case 3:
+                return boost::get<std::string>(value);
+        }
+        LOG(FATAL) << "unknown type: " << value.which();
+    }
+
+    static bool toBool(const VariantType &value) {
+        return asBool(value);
+    }
+
+    static double toDouble(const VariantType &value) {
+        switch (value.which()) {
+            case 0:
+                return static_cast<double>(boost::get<int64_t>(value));
+            case 1:
+                return boost::get<double>(value);
+            case 2:
+                return boost::get<bool>(value) ? 1.0 : 0.0;
+            case 3:
+                // TODO(dutor) error handling
+                return folly::to<double>(boost::get<std::string>(value));
```

Comment:
I think folly::tryTo would be better here, or exception would occur if an unparseable string is provided.

Reply:
I will address such problems after #669 is merged.